### PR TITLE
[REF] Simplify getContributionStatuses

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
[REF] Simplify getContributionStatuses

The existing function is really confusing - but all that confusion is designed for one
form - back office contribution form - and augmented by overloading by other functions.

This splits it out. The same options are returned - however if sites have added their own options they would not be returned - we don't support that in the UI and our financial code really does expect a limited range of statuses so I don't know if that is a thing - maybe highlight in release notes?

Before
----------------------------------------
Function is super hard to read - being called to do different things makes that hard to unwravel. The function is used to get the drop down options for available statuses in 5 scenarions

1) new contribution backoffice (available options are Pending, Completed, Failed, Cancelled)
image
2) new membership with record payment (available options are Pending & Completed)
3) new participant with record payment (available options are Pending & Completed)
4) Profile form with contribution status included  (available options are Pending, Completed, Failed, Cancelled)
5) edit contribution back-office - this one is all over the show and untouched by this PR

After
----------------------------------------
The first 4 scenarios call simpler functions - leaving the last to be fixed up.

Technical Details
----------------------------------------
Could sites be using 'weird and wonderful other statuses'? Our system really is set up
around supporting specific statuses and doing financial entries for those so I think maybe
we are right to not try to pander to any that could exist

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins re test this please

File: CRM/Event/Form/Participant.php [hunk 1]
```diff
@@ -1733,7 +1733,7 @@ public function buildEventFeeForm($form) {
         }
 
         $form->add('select', 'contribution_status_id',
-          ts('Payment Status'), CRM_Contribute_BAO_Contribution_Utils::getContributionStatuses('participant')
+          ts('Payment Status'), CRM_Contribute_BAO_Contribution_Utils::getPendingAndCompleteStatuses('participant')
```
Comment (demeritcowboy) [inline CRM/Event/Form/Participant.php:1736]: I'm still trying to get my head around the earlier function but this function doesn't take a parameter here.

File: CRM/Member/Form/MembershipRenewal.php [hunk 2]
```diff
@@ -359,7 +359,7 @@ public function buildQuickForm() {
       );
 
       $this->add('select', 'contribution_status_id', ts('Payment Status'),
-        CRM_Contribute_BAO_Contribution_Utils::getContributionStatuses('membership')
+        CRM_Contribute_BAO_Contribution_Utils::getPendingAndCompleteStatuses('membership')
```
Comment (demeritcowboy) [inline CRM/Member/Form/MembershipRenewal.php:362]: Ditto - no parameter

File: CRM/Event/Form/Participant.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Event/Form/Participant.php:1736]: oh yeah - that was lazy of me

**Comment by demeritcowboy** [comment]
Jenkins retest this please - `E2E\Core\AssetBuilderTest.testInvalid`

**Comment by demeritcowboy** [comment]
Good job unravelling that function. Contribution status = "better".

**Comment by eileenmcnaughton** [comment]
Just noting I was going to remove the `$usedFor` param but found a couple of universe calls so I re-instated it & deprecated the function in #22345

**Pull request merged**